pkgs/applications/terminal-emulators/cool-retro-term/default.nix

```diff
@@ -1,22 +1,22 @@
 { lib, stdenv, fetchFromGitHub, mkDerivation, qtbase, qtquick1, qmltermwidget
-, qtquickcontrols, qtgraphicaleffects, qmake, nixosTests }:
+, qtquickcontrols2, qtgraphicaleffects, qmake, nixosTests }:
 
 mkDerivation rec {
-  version = "1.1.1";
+  version = "1.2.0";
   pname = "cool-retro-term";
 
   src = fetchFromGitHub {
     owner = "Swordfish90";
     repo = "cool-retro-term";
     rev = version;
-    sha256 = "0mird4k88ml6y61hky2jynrjmnxl849fvhsr5jfdlnv0i7r5vwi5";
+    sha256 = "PewHLVmo+RTBHIQ/y2FBkgXsIvujYd7u56JdFC10B4c=";
   };
 
   patchPhase = ''
     sed -i -e '/qmltermwidget/d' cool-retro-term.pro
   '';
```

Comment:
I would replace this with the following as a patch file. That way the intent is clear and other patch files can be added. Replacing phases completely like this causes problems in overrides.

`use-system-qmltermwidget.patch`

```
diff --git a/cool-retro-term.pro b/cool-retro-term.pro
index 851f0b4..b24c473 100644
--- a/cool-retro-term.pro
+++ b/cool-retro-term.pro
@@ -2,7 +2,6 @@ TEMPLATE = subdirs
 
 CONFIG += ordered
 
-SUBDIRS += qmltermwidget
 SUBDIRS += app
 
 desktop.files += cool-retro-term.desktop
```

Reply:
Refactored as suggested; thank you.